packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -37,13 +37,10 @@ export class DataSourceWithBackend<
    * Ideally final -- any other implementation may not work as expected
    */
   query(request: DataQueryRequest): Observable<DataQueryResponse> {
-    const { targets, intervalMs, maxDataPoints, range } = request;
-
-    let expressionCount = 0;
+    const { targets, intervalMs, maxDataPoints, range, requestId } = request;
     const orgId = config.bootData.user.orgId;
     const queries = targets.map(q => {
       if (q.datasource === ExpressionDatasourceID) {
-        expressionCount++;
```

Comment:
Is this hack not needed anymore?  If not where are the backend changes that go with it?

Reply:
Doesn't look like it. The `expressionCount` was provided in body on root level but the grafana backend didn't receive that at all.